Parsers/ASimFileEvent/Parsers/vimFileEventVMwareCarbonBlackCloud.yaml

```diff
@@ -0,0 +1,150 @@
+ParserName: vimFileEventVMwareCarbonBlackCloud
+EquivalentBuiltInParser: _Im_FileEvent_VMwareCarbonBlackCloud
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventFieldsLookup = datatable(
+      sensor_action_s: string,
+      DvcAction: string,
+      EventResult: string
+  )[
+      "ACTION_ALLOW", "Allow", "Success",
+      "ACTION_BLOCK", "Block", "Failure",
+      "ACTION_TERMINATE", "Terminate", "Failure",
+      "ACTION_BREAK", "Break", "Failure",
+      "ACTION_SUSPEND", "Suspend", "Failure",
+      "", "", "Success"
+  ];
+  let EventTypeLookup = datatable(action_s: string, EventType: string)[
+      "ACTION_FILE_CREATE", "FileCreated",
+      "ACTION_FILE_DELETE", "FileDeleted",
+      "ACTION_FILE_LAST_WRITE", "FileModified",
+      "ACTION_FILE_LINK", "FileModified",
+      "ACTION_FILE_READ", "FileAccessed",
+      "ACTION_FILE_RENAME", "FileRenamed",
+      "ACTION_FILE_WRITE", "FileModified",
+      "ACTION_FILE_OPEN_DELETE", "FileDeleted",
+      "ACTION_FILE_OPEN_EXECUTE", "FileAccessed",
+      "ACTION_FILE_OPEN_SET_ATTRIBUTES", "FileAttributesUpdated",
+      "ACTION_FILE_OPEN_SET_SECURITY", "FileAttributesUpdated",
+      "ACTION_FILE_SET_SECURITY", "FileAttributesUpdated",
+      "ACTION_FILE_TRUNCATE", "FileModified",
+      "ACTION_FILE_OPEN_WRITE", "FileModified",
+      "ACTION_FILE_MOD_OPEN", "FileAccessed",
+      "ACTION_FILE_OPEN_READ", "FileAccessed"
+  ];
+  let parser = (disabled: bool=false) {
+      CarbonBlackEvents_CL
+      | where eventType_s == "endpoint.event.filemod" and isnotempty(filemod_name_s)
+      | where action_s !in ("ACTION_INVALID", "ACTION_FILE_UNDELETE")
+      | parse filemod_hash_s with * '["' TargetFileMD5: string '","' TargetFileSHA256: string '"]'
+      | lookup EventFieldsLookup on sensor_action_s
+      | extend temp_action = iff(action_s has "|", action_s, "")
+      | lookup EventTypeLookup on action_s
+      | extend EventType = case(
+                          isnotempty(EventType), EventType,
+                          temp_action has "delete", "FileDeleted",
+                          temp_action has "link", "FileModified",
+                          temp_action has "rename", "FileRenamed",
+                          temp_action has "execute", "FileAccessed",
+                          temp_action has_any ("attributes", "security"), "FileAttributesUpdated",
+                          temp_action has "truncate", "FileModified",
+                          temp_action has "write", "FileModified",
+                          temp_action has_any ("read", "open"), "FileAccessed",
+                          temp_action has "create", "FileCreated",
+                          ""
```

Comment:
why we planned of using both lookup table and case for EventType, isn't line 66 going to override the updates from line 65? With this seems temp_action value is prioritized which is also evaluatd from action_s

Reply:
As we are getting multiple values in the action_s field many times, we have define priority for them and set values in temp_action and accordingly, we are setting EventType value.
EventType will be set based on lookup only for records where a single value is present in the action_s field and it won't be overridden as we have this condition {isnotempty(EventType), EventType} as first priority.